central/reports/scheduler/schedule_test.go

```diff
@@ -36,59 +37,71 @@ type vulnsAndDate struct {
 	DateStr    string
 }
 
-func generateExpectedVulnReportEmailTemplates(t *testing.T) (string, string) {
+func TestBrandedText(t *testing.T) {
+	suite.Run(t, new(ScheduleTestSuite))
+}
+
+type ScheduleTestSuite struct {
+	suite.Suite
+
+	rc                                              *storage.ReportConfiguration
+	envIsolator                                     *envisolator.EnvIsolator
+	expectedVulnReportEmailTemplateRhacsBranding    string
+	expectedVulnReportEmailTemplateStackroxBranding string
+}
+
+func (s *ScheduleTestSuite) SetupSuite() {
+	s.envIsolator = envisolator.NewEnvIsolator(s.T())
+	s.rc = fixtures.GetValidReportConfiguration()
+
 	data := &vulnsAndDate{
 		WhichVulns: "for all vulnerabilities",
 		DateStr:    timeStr,
 	}
 
 	tmpl, err := template.New("VulnsRHACS").Parse(expectedVulnReportEmailTemplateRhacsBrandingWithPlaceholders)
-	assert.NoError(t, err)
-	expectedVulnReportEmailTemplateRhacsBranding, err := templates.ExecuteToString(tmpl, data)
-	assert.NoError(t, err)
+	s.NoError(err)
+	s.expectedVulnReportEmailTemplateRhacsBranding, err = templates.ExecuteToString(tmpl, data)
+	s.NoError(err)
 
 	tmpl, err = template.New("VulnsStackrox").Parse(expectedVulnReportEmailTemplateStackroxBrandingWithPlaceholders)
-	assert.NoError(t, err)
-	expectedVulnReportEmailTemplateStackroxBranding, err := templates.ExecuteToString(tmpl, data)
-	assert.NoError(t, err)
-
-	return expectedVulnReportEmailTemplateRhacsBranding, expectedVulnReportEmailTemplateStackroxBranding
+	s.NoError(err)
+	s.expectedVulnReportEmailTemplateStackroxBranding, err = templates.ExecuteToString(tmpl, data)
+	s.NoError(err)
 }
 
-func TestFormatVulnMessage(t *testing.T) {
-	envIsolator := envisolator.NewEnvIsolator(t)
-	rc := fixtures.GetValidReportConfiguration()
-
-	expectedVulnReportEmailTemplateRhacsBranding, expectedVulnReportEmailTemplateStackroxBranding := generateExpectedVulnReportEmailTemplates(t)
+func (s *ScheduleTestSuite) TeardownTest() {
+	s.envIsolator.RestoreAll()
+}
 
+func (s *ScheduleTestSuite) TestFormatVulnMessage() {
 	tests := map[string]struct {
 		productBranding string
 		vulnReport      string
 		noVulnReport    string
 	}{
 		"RHACS branding": {
 			productBranding: "RHACS_BRANDING",
-			vulnReport:      expectedVulnReportEmailTemplateRhacsBranding,
+			vulnReport:      s.expectedVulnReportEmailTemplateRhacsBranding,
 			noVulnReport:    expectedNoVulnsFoundEmailTemplateRhacsBranding,
 		},
 		"StackRox branding": {
 			productBranding: "STACKROX_BRANDING",
-			vulnReport:      expectedVulnReportEmailTemplateStackroxBranding,
+			vulnReport:      s.expectedVulnReportEmailTemplateStackroxBranding,
 			noVulnReport:    expectedNoVulnsFoundEmailTemplateStackroxBranding,
 		},
 	}
 	for name, tt := range tests {
-		t.Run(name, func(t *testing.T) {
-			envIsolator.Setenv(branding.ProductBrandingEnvName, tt.productBranding)
+		s.T().Run(name, func(t *testing.T) {
```

Comment:
```suggestion
		s.Run(name, func() {
```

Reply:
`suite` provides a wrapper for subtests where you don't need to provide the inner `t` parameter.

Reply:
Nice, I'll use that.